doppler-env/src/main.ts

```diff
@@ -0,0 +1,38 @@
+import { exec } from 'child_process';
+import fs from 'fs/promises';
+import * as core from '@actions/core';
+
+const main = async (): Promise<void> => {
+  const releaseStage = core.getInput('releaseStage');
+  const tokenKey = `${releaseStage.toUpperCase()}_DOPPLER_TOKEN`;
+  const token = process.env[tokenKey];
+
+  if (!token) {
+    throw new Error(`No token for ${tokenKey} in environment`);
+  }
+
+  await exec(`doppler secrets download --no-file --format=json --token=${token} > temp.json`);
+
+  const secretsFile = await fs.readFile(`${process.env.GITHUB_WORKSPACE}/temp.json`, 'utf8');
+  const easJsonFile = await fs.readFile(`${process.env.GITHUB_WORKSPACE}/eas.json`, 'utf8');
+  const secrets = JSON.parse(secretsFile);
+  const easConfig = JSON.parse(easJsonFile);
+
+  const updatedEasJson = {
+    ...easConfig,
+    build: {
+      ...easConfig.build,
+      [releaseStage]: {
+        ...easConfig.build[releaseStage],
+        ...secrets,
+      },
+    },
+  };
```

Comment:
Per or external convo: 
Trying to avoid using Expo/Constants (one less dependency - requires Expo Modules) 

- Experiment with Bable to replace ENV variables 
https://docs.expo.dev/guides/environment-variables/#using-babel-to-replace-variables
- Assuming this works, we don't need to write an EAS file (Can commit flat file without "env") 
- Doppler to write to `.env` file 
PRO: This removes all the MOBILE specifics 
-- Action can now be used for Web/Mobile 
-- Can create doc examples for both

CON: Bable plugin reliant 


**Option 2:**
If we need to write to `app.config` 
- Look into `inline dotenv` 
https://docs.expo.dev/guides/environment-variables/#using-a-dotenv-file

PRO: 
- Doppler Action can still be be built for Web/Mobile writing to a `.env` 
- Leverage only the ENVs needed in app.config.


PRO FOR BOTH OPTIONS: 
- If a client decided to not use DOTENV, they would manage ENVs via the Expo dashboard. These ENVs __SHOULD__ pull through the same way given the babel/dotenv configs noted above.